src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/allocation/AttemptAllocationStep.kt

```diff
@@ -0,0 +1,73 @@
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.step.allocation
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.AllocationActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.managedindexmetadata.StepMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.Step
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.admin.indices.settings.put.UpdateSettingsRequest
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.settings.Settings
+
+class AttemptAllocationStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val config: AllocationActionConfig,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step("attempt_allocation", managedIndexMetaData) {
+    private val logger = LogManager.getLogger(javaClass)
+    private var stepStatus = StepStatus.STARTING
+    private var info: Map<String, Any>? = null
+
+    override fun isIdempotent() = true
+
+    override suspend fun execute() {
+        try {
+            val response: AcknowledgedResponse = client.admin()
+                .indices()
+                .suspendUntil { updateSettings(UpdateSettingsRequest(buildSettings(), managedIndexMetaData.index), it) }
+            handleResponse(response)
+        } catch (e: Exception) {
+            logger.error(ERROR_MESSAGE, e)
+            stepStatus = StepStatus.FAILED
+            val mutableInfo = mutableMapOf("message" to ERROR_MESSAGE)
+            val errorMessage = e.message
+            if (errorMessage != null) mutableInfo["cause"] = errorMessage
+            info = mutableInfo.toMap()
+        }
+    }
+
+    private fun buildSettings(): Settings {
+        val builder = Settings.builder()
+        config.require.forEach { (key, value) -> builder.put(SETTINGS_PREFIX + AllocationActionConfig.REQUIRE + "." + key, value) }
```

Comment:
Can the key values be absolutely anything? Or should we have some validation on them in the parser? i.e. can you have something like `"index.routing.allocation._?__": " "` or will it throw an exception?

Reply:
While it is not possible to configure key like `"index.routing.allocation._..__": " "` it is possible to configure `"index.routing.allocation._._._._.": " "` which is equally useless. As far as I know elastic doesn't provide one regex that we can match to so I've added a test that shows that any invalid key will result in a failed policy.